eth/beacon/helpers.py

```diff
@@ -408,38 +408,77 @@ def get_domain(fork_data: 'ForkData',
     ) * 4294967296 + domain_type
 
 
-def verify_slashable_vote_data(state: BeaconState,
-                               vote_data: 'SlashableVoteData',
-                               max_casper_votes: int) -> bool:
+@to_tuple
+def get_pubkey_for_indices(validators: Sequence[ValidatorRecord],
+                           indices: Sequence[int]) -> Iterable[int]:
+    for index in indices:
+        yield validators[index].pubkey
+
+
+@to_tuple
+def generate_aggregate_pubkeys(state: BeaconState, vote_data: 'SlashableVoteData') -> Iterable[int]:
+    """
+    Computes the aggregate pubkey we expect based on
+    the proof-of-custody indices found in the ``vote_data``.
+    """
     proof_of_custody_0_indices = vote_data.aggregate_signature_poc_0_indices
     proof_of_custody_1_indices = vote_data.aggregate_signature_poc_1_indices
-    vote_count = len(proof_of_custody_0_indices) + len(proof_of_custody_1_indices)
-    if vote_count > max_casper_votes:
-        return False
+    all_indices = (proof_of_custody_0_indices, proof_of_custody_1_indices)
+    get_pubkeys = functools.partial(get_pubkey_for_indices, state.validators)
+    return map(bls.aggregate_pubkeys,
+               map(get_pubkeys, all_indices))
 
```

Comment:
general note: i'll lean towards a functional style where i can -- if any contributor wants to step in and nudge me towards what some consider a more pythonic style, i'm all ears :) i think it's important to follow the established repo guidelines but will bend the rules where i can to get what imo is more expressive code ;)

Reply:
https://github.com/ethereum/snake-charmers-tactical-manual/blob/master/style-guide.md

If you haven't seen that yet it outlines a lot of our style guides but we also are all about rule bending when the situation calls for it.

:heart: functional style